pkg/db/statements/sqls.go

```diff
@@ -3,23 +3,78 @@ package statements
+	"fmt"
+	"strings"
 )
 
 //go:embed *.sql
 var fs embed.FS
 
-func (s *Statements) CreateSQL() string           { return s.statements["migrate.sql"] }
-func (s *Statements) InsertSQL() string           { return s.statements["insert.sql"] }
-func (s *Statements) TableMetaSQL() string        { return s.statements["tablemeta.sql"] }
-func (s *Statements) ClearCreatedSQL() string     { return s.statements["clearcreated.sql"] }
+func (s *Statements) CreateSQL() string { return s.statements["migrate.sql"] }
+
+func (s *Statements) AddColumnSQL(name string) string {
+	return strings.Replace(s.statements["addcolumn.sql"], "new_column", strings.ReplaceAll(name, ".", "_"), 1)
+}
+
+func (s *Statements) AddFieldsIndexSQL(fields []string) string {
+	var fieldsToIndex string
+	for _, f := range fields {
+		if f != "" && !strings.HasPrefix(f, "metadata.name") {
+			fieldsToIndex += fmt.Sprintf(", %s", strings.ReplaceAll(f, ".", "_"))
+		}
+	}
+	if fieldsToIndex == "" {
+		return ""
+	}
+
+	return strings.Replace(s.statements["addfieldsindex.sql"], "extra_fields", fieldsToIndex, 1)
+}
+
+func (s *Statements) DropFieldsIndexSQL() string { return s.statements["dropfieldsindex.sql"] }
+
+func (s *Statements) InsertSQL(fieldNames []string) string {
+	var extraFields, extraVals string
+	for i, f := range fieldNames {
+		if !strings.HasPrefix(f, "metadata.") {
+			extraFields += fmt.Sprintf(", %s", strings.ReplaceAll(f, ".", "_"))
+			extraVals += fmt.Sprintf(",$%d", i+8)
+		}
+	}
+	return strings.Replace(strings.Replace(s.statements["insert.sql"], "extra_fields", extraFields, 1), "extra_vals", extraVals, 1)
+}
+
+func (s *Statements) TableMetaSQL() string { return s.statements["tablemeta.sql"] }
+
+func (s *Statements) ClearCreatedSQL() string { return s.statements["clearcreated.sql"] }
+
 func (s *Statements) UpdateCompactionSQL() string { return s.statements["updatecompaction.sql"] }
-func (s *Statements) CompactSQL() string          { return s.statements["compact.sql"] }
-func (s *Statements) listSQL() string             { return s.statements["list.sql"] }
-func (s *Statements) listAfterSQL() string        { return s.statements["listafter.sql"] }
+
+func (s *Statements) CompactSQL() string { return s.statements["compact.sql"] }
+
+func (s *Statements) listSQL(extraFields []string) string {
+	return strings.Replace(s.statements["list.sql"], "extra_fields", extraFieldsWithIndexOffset(extraFields, 5), 1)
+}
+
+func (s *Statements) listAfterSQL(extraFields []string) string {
+	return strings.Replace(s.statements["listafter.sql"], "extra_fields", extraFieldsWithIndexOffset(extraFields, 4), 1)
+}
 
 func (s *Statements) TableLockSQL() string {
 	if s.lock {
 		return s.statements["tablelock.sql"]
 	}
 	return ""
 }
+
+func extraFieldsWithIndexOffset(extraFields []string, offset int) string {
+	var extraFieldsStr string
+	for i, f := range extraFields {
+		// metadata.namespace and metadata.name are handled elsewhere.
+		if !strings.HasPrefix(f, "metadata.name") {
```

Comment:
oh you cheeky bastard, prefix on metadata.name. I guess it's technically correct, but obscure.